src/vos/vos_ilog.c

```diff
@@ -63,10 +63,13 @@ vos_ilog_is_same_tx(struct umem_instance *umm, uint32_t tx_id,
 {
 	uint32_t dtx = vos_dtx_get();
 
-	/* If we are not in a dtx, treat the in-tree entry as
-	 * if it's a different transaction.
+	/* If the entry is committed, treat the new update as a different
+	 * transaction.   If the operation (update or punch) is the same,
+	 * we can detect this and return -DER_ALREADY.   If it's not,
+	 * we require either a major or minor epoch change and will
+	 * return -DER_TX_RESTART.
```

Comment:
Since minor epoch is only used inside VOS, then if they are not the same, we only can require major epoch to be changed via return -DER_TX_RESTART, right?

Reply:
This is only an issue if one sends two updates with same major epoch.   Minor epoch is internal to vos and for vos, this rc means needing a major update.  But for the ilog test, we can actually update the minor epoch